streamalert_cli/manage_lambda/deploy.py

```diff
@@ -159,80 +153,101 @@ def _update_rule_table(options, config):
-            {'module.threat_intel_downloader'},
-            config['lambda'].get('threat_intel_downloader_config', False)
-        )
+    target_mapping = {
+        'alert': {
+            'targets': {
+                'module.alert_processor_iam',
+                'module.alert_processor_lambda',
+            },
+            'enabled': True  # required function
+        },
+        'alert_merger': {
+            'targets': {
+                'module.alert_merger_iam',
+                'module.alert_merger_lambda',
+            },
+            'enabled': True  # required function
+        },
+        'athena': {
+            'targets': {
+                'module.athena_partitioner_iam',
+                'module.athena_partitioner_lambda',
+            },
+            'enabled': True  # required function
+        },
+        'rule': {
+            'targets': {
+                'module.rules_engine_iam',
+                'module.rules_engine_lambda',
+            },
+            'enabled': True  # required function
+        },
+        'classifier': {
+            'targets': {
+                'module.classifier_{}_{}'.format(cluster, suffix)
+                for suffix in {'lambda', 'iam'}
+                for cluster in clusters
+            },
+            'enabled': bool(clusters)  # one cluster at least is required
+        },
+        'apps': {
+            'targets': {
+                'module.app_{}_{}_{}'.format(app_info['app_name'], cluster, suffix)
+                for suffix in {'lambda', 'iam'}
+                for cluster in clusters
+                for app_info in config['clusters'][cluster]['modules'].get(
+                    'streamalert_apps', {}
+                ).values()
+                if 'app_name' in app_info
+            },
+            'enabled': any(
+                info['modules'].get('streamalert_apps')
+                for info in config['clusters'].values()
+            )
+        },
+        'rule_promo': {
+            'targets': {
+                'module.rule_promotion_iam',
+                'module.rule_promotion_lambda',
+            },
+            'enabled': config['lambda'].get('rule_promotion_config', {}).get('enabled', False)
+        },
+        'scheduled_queries': {
+            'targets': {
+                'module.scheduled_queries',
+            },
+            'enabled': config['scheduled_queries'].get('enabled', False),
+        },
+        'threat_intel_downloader': {
+            'targets': {
+                'module.threat_intel_downloader',
+                'module.threat_intel_downloader_iam',
+            },
+            'enabled': config['lambda'].get('threat_intel_downloader_config', False),
+        }
     }
 
-    if not package_mapping[function_name].enabled:
-        return False, False
+    targets = set()
+    for function in functions:
+        if not target_mapping[function]['enabled']:
+            LOGGER.error('Function is not enabled and will be ignored: %s', function)
```

Comment:
Should this be an error or just a warning since it looks like it continues even if the "error" occurs.

Reply:
good point - made it warning